cmd/agent/app/agent.go

```diff
@@ -86,8 +88,20 @@ func (a *Agent) HTTPAddr() string {
 
 // Stop forces all agent go routines to exit.
 func (a *Agent) Stop() {
+	// first, close the http server, so that we don't have any more inflight requests
```

Comment:
```suggestion
	// first, close the http server, so that we don't have any more inflight requests
	a.logger.Info("shutting down agent's HTTP server", zap.String("addr", a.httpAddr.Load()))
```